src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -94,13 +94,7 @@ internal sealed class TupleTypeSymbol : WrappedNamedTypeSymbol
         {
             Debug.Assert(!shouldCheckConstraints || (object)syntax != null);
             Debug.Assert(elementNames.IsDefault || elementTypes.Length == elementNames.Length);
-
-            int numElements = elementTypes.Length;
-
-            if (numElements <= 1)
-            {
-                throw ExceptionUtilities.Unreachable;
-            }
+            Debug.Assert(elementTypes.Length > 1 || elementNames.IsDefault);
```

Comment:
:memo: This can now be reached (see `TestOn1Tuple`), when we need construct converted types for `ValueTuple'1`.
I'm not 100% confident that's the proper way to handle.